api/src/main/java/org/openmrs/module/sync2/api/impl/SyncConfigurationServiceImpl.java

```diff
@@ -11,71 +25,55 @@
-import static org.openmrs.module.sync2.api.utils.SyncConfigurationUtils.customConfigExists;
-import static org.openmrs.module.sync2.api.utils.SyncConfigurationUtils.parseJsonFileToSyncConfiguration;
-import static org.openmrs.module.sync2.api.utils.SyncConfigurationUtils.writeSyncConfigurationToJsonFile;
-import static org.openmrs.module.sync2.api.utils.SyncConfigurationUtils.isValidateJson;
-import static org.openmrs.module.sync2.api.utils.SyncConfigurationUtils.parseJsonStringToSyncConfiguration;
-
 @Component("sync2.syncConfigurationService")
 public class SyncConfigurationServiceImpl implements SyncConfigurationService {
 
-    private SyncConfiguration syncConfiguration;
-
-    @Autowired
-    private SyncSchedulerService schedulerService;
-
-    public SyncConfigurationServiceImpl() {
-        String configFilePath = getConfigFilePath();
-
-        this.syncConfiguration = customConfigExists(configFilePath) ?
-                parseJsonFileToSyncConfiguration(configFilePath, ABSOLUTE) :
-                parseJsonFileToSyncConfiguration(SYNC2_PATH_TO_DEFAULT_CONFIGURATION, RELATIVE);
-    }
+	@Autowired
+	private SyncSchedulerService schedulerService;
 
-    @Override
-    public void saveConfiguration(SyncConfiguration configuration) throws SyncException {
-        writeSyncConfigurationToJsonFile(configuration, getConfigFilePath());
-        this.syncConfiguration = configuration;
-        schedulerService.runSyncScheduler();
-    }
+	@Override
+	public SyncConfiguration getSyncConfiguration() {
+		String configFilePath = getConfigFilePath();
 
-    @Override
-    public void saveConfiguration(String jsonConfiguration) {
-        if (isValidateJson(jsonConfiguration)) {
-            SyncConfiguration customConfiguration = parseJsonStringToSyncConfiguration(jsonConfiguration);
-            writeSyncConfigurationToJsonFile(customConfiguration, getConfigFilePath());
-            this.syncConfiguration = customConfiguration;
+		return customConfigExists(configFilePath)
+				? parseJsonFileToSyncConfiguration(configFilePath, ABSOLUTE)
+				: parseJsonFileToSyncConfiguration(
+						SYNC2_PATH_TO_DEFAULT_CONFIGURATION, RELATIVE);
+	}
 
-            schedulerService.runSyncScheduler();
-        }
-    }
+	@Override
+	public void saveConfiguration(SyncConfiguration configuration)
+			throws SyncException {
+		writeSyncConfigurationToJsonFile(configuration, getConfigFilePath());
+		schedulerService.runSyncScheduler();
+	}
 
-    @Override
-    public SyncConfiguration getSyncConfiguration() {
-        return this.syncConfiguration;
-    }
+	@Override
+	public void saveConfiguration(String jsonConfiguration) {
+		if (isValidateJson(jsonConfiguration)) {
+			SyncConfiguration customConfiguration = parseJsonStringToSyncConfiguration(
+					jsonConfiguration);
+			writeSyncConfigurationToJsonFile(customConfiguration,
+					getConfigFilePath());
+			schedulerService.runSyncScheduler();
```

Comment:
I think we should stick to "this.fieldName" syntax to reference fields.

Reply:
thanks @ubap, am not sure i understand this well enough, could you clarify!